### PR TITLE
Some cheatcodes work on test state

### LINKED ISSUE #612: Cheatcodes should work in test state
If we target address `0`, cheatcode should target test state

### PR DESCRIPTION
**Pull request by abulenok**
Partially solves #612 

## Introduced changes

- roll, warp and prank work on a test state
-

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by war-in — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

File: crates/forge/tests/integration/test_state.rs [hunk 1]
```diff
@@ -331,3 +331,105 @@ fn test_cant_call_test_contract() {
...
+    "#
+    ));
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+#[ignore] //TODO (#612)
```
Comment (cptartur) [inline crates/forge/tests/integration/test_state.rs:403]: Either the description of this PR should be changed so #612 is not closed after merge of this, or we should create an separate issue for this test.

**Review by cptartur — APPROVED** [review]

File: crates/forge/tests/integration/test_state.rs (hunk 1 above)
Comment (abulenok) [reply crates/forge/tests/integration/test_state.rs:403]: Updated the description, so this PR will not close the issue

**Review by piotmag769 — APPROVED** [review]

File: crates/cheatnet/src/execution/syscalls.rs [hunk 2]
```diff
@@ -109,13 +109,13 @@ pub fn call_contract_syscall(
     // endregion
 }
 
-pub struct CheatableSyscallHandler<'a> {
-    pub syscall_handler: SyscallHintProcessor<'a>,
+pub struct CheatableSyscallHandler<'a, 'b> {
+    pub syscall_handler: &'a mut SyscallHintProcessor<'b>,
```
[the remark this answers is not in the document]
Comment (abulenok) [reply crates/cheatnet/src/execution/syscalls.rs:113]: The compiler will protest eagerly

**Comment by MaksymilianDemitraszek** [comment]
I will be reviewing this one please don't merge

**Comment by MaksymilianDemitraszek** [comment]
This one will be probably solved by this PR https://github.com/foundry-rs/starknet-foundry/pull/753

It makes behaviour consistent between syscall handlers

File: crates/forge/tests/integration/test_state.rs [hunk 3]
```diff
@@ -391,3 +391,102 @@ fn test_cant_call_test_contract() {
...
```
Comment (piotmag769) [inline crates/forge/tests/integration/test_state.rs:463]: Why doesn't it work?

**Review by piotmag769 — APPROVED** [review]

File: crates/forge/tests/integration/test_state.rs (hunk 3 above)
Comment (war-in) [reply crates/forge/tests/integration/test_state.rs:463]: It should now

Comment (abulenok) [reply crates/forge/tests/integration/test_state.rs:463]: this will be resolved in a separate PR

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]
LGTM

**Pull request merged**